ambry-clustermap/src/main/java/com.github.ambry.clustermap/HelixClusterManager.java

```diff
@@ -373,10 +376,11 @@ private void updateSealedStateOfReplicas(List<InstanceConfig> instanceConfigs) {
     @Override
     public void onLiveInstanceChange(List<LiveInstance> liveInstances, NotificationContext changeContext) {
       logger.trace("Live instance change triggered from {} with: {}", dcName, liveInstances);
-      if (changeContext.getType() == NotificationContext.Type.INIT) {
+      updateInstanceLiveness(liveInstances);
+      if (!liveStateInitialized) {
```

Comment:
This is intentionally not atomic since the initial callback comes from within the registration call itself before subsequent callbacks, so this variable never gets updated concurrently. I will make it atomic to be safe.